pkg/cmd/join/exec.go

```diff
@@ -92,6 +95,16 @@ func (o *Options) validate() error {
 		return fmt.Errorf("registry should not be empty")
 	}
 
+	// preflight check
+	if err := preflightinterface.RunChecks(
+		[]preflightinterface.Checker{
+			preflight.KlusterletApiserverCheck{
```

Comment:
are you sure it works, we read and set this in complete. I think validate is run before complete?

Reply:
@qiujian16 seems complete runs first, then validate runs: https://github.com/open-cluster-management-io/clusteradm/blob/main/pkg/cmd/join/cmd.go#L33

Reply:
hrm, with that I think we need to do preflight in complete. The message here is not correct, because apiserver can also be inputed by user I think?